build_system/src/rust_tools.rs

```diff
@@ -97,6 +96,22 @@ impl RustcTools {
     }
 }
 
+fn exec(input: &[&dyn AsRef<OsStr>], env: &HashMap<String, String>) -> Result<(), String> {
+    #[cfg(unix)]
+    {
+        let error = crate::utils::get_command_inner(input, None, Some(env)).exec();
+        eprintln!("Command failed: {error:?}");
```

Comment:
I would rephrase the error to make this clear that it's the `execvp` syscall that failed.